src/native/runtime-base/android-system.cc

```diff
@@ -262,26 +262,14 @@ AndroidSystem::monodroid_get_system_property_from_overrides ([[maybe_unused]] co
 	return 0;
 }
 
-// TODO: review this. Do we really have to create the dir in release?
 void
 AndroidSystem::create_update_dir (char *override_dir) noexcept
 {
-#if defined (RELEASE)
-	/*
-	 * Don't create .__override__ on Release builds, because Google requires
-	 * that pre-loaded apps not create world-writable directories.
-	 *
-	 * However, if any logging is enabled (which should _not_ happen with
-	 * pre-loaded apps!), we need the .__override__ directory...
-	 */
-	if (log_categories == 0 && monodroid_get_system_property (SharedConstants::DEBUG_MONO_PROFILE_PROPERTY, nullptr) == 0) {
-		return;
-	}
-#endif // def RELEASE
-
 	override_dirs [0] = override_dir;
+#if defined(DEBUG)
+	log_debug (LOG_DEFAULT, "Creating public update directory: `%s`", override_dir);
 	Util::create_public_directory (override_dir);
```

Comment:
There are some cases we need to create the `.__override__` directory in release mode that uses `libmono-android.release.so`:
* Recording an AOT profile saves the `custom.aprof` file here
* [JIT times](https://github.com/dotnet/android/blob/main/Documentation/guides/profiling.md#profiling-the-jit-compiler) saves `methods.txt` here

These changes look like it would only create `.__override__` in `libmono-android.debug.so`? Is that the case?

Reply:
The other situation, for release build profiling, is handled here: https://github.com/dotnet/android/pull/9367/files/01677765e2d9f55b46b96aa43d24ffc50b430115#diff-5296758f320149c3439879b0262425a13756cddabe72b83f286a5ebfc392ec7fR1127

Reply:
@grendello that link doesn't work? I think the commit was rebased

Reply:
Oof, indeed, here's one which should work https://github.com/dotnet/android/blob/8dfab2e743c0e9c5c67decf8de909ffcc9f37d0f/src/native/monodroid/monodroid-glue.cc#L1127